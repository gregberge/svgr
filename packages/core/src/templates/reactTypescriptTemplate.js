export default (code, opts = {}, state) => {
  let props = ''

  if (opts.expandProps && opts.ref && opts.titleProp) {
    props = '{svgRef: string, title: string, ...props}'
  } else if (opts.expandProps && opts.titleProp) {
    props = '{title: string, ...props}'
  } else if (opts.expandProps && opts.ref) {
    props = '{svgRef: string, ...props}'
  } else if (opts.titleProp && opts.ref) {
    props = '{svgRef: string, title: string}'
  } else if (opts.expandProps) {
    props = 'props: any'
  } else if (opts.ref) {
    props = '{svgRef: string}'
  } else if (opts.titleProp) {
    props = '{title: string}'
  }

  let result = `import * as React from 'react'\n\n`
  result += `const ${state.componentName} = (${props}) => ${code}\n\n`

  if (state.webpack && state.webpack.previousExport) {
    result += `export default ${state.webpack.previousExport}\n`
    result += `export { ${state.componentName} as ReactComponent }`
  } else if (state.rollup && state.rollup.previousExport) {
    result += `${state.rollup.previousExport}\n`
    result += `export { ${state.componentName} as ReactComponent }`
  } else {
    result += `export default ${state.componentName}`
  }

  return result
}
