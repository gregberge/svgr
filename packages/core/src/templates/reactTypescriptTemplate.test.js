import reactTypescriptTemplate from './reactTypescriptTemplate'

const t = (options, state = { componentName: 'Test' }) =>
  reactTypescriptTemplate('<Svg />', options, state)

describe('reactTypescriptTemplate', () => {
  it('should wrap code into a component', () => {
    expect(t({})).toMatchSnapshot()
    expect(t({ titleProp: true })).toMatchSnapshot()
    expect(t({ ref: true })).toMatchSnapshot()
    expect(t({ expandProps: true })).toMatchSnapshot()
    expect(t({ titleProp: true, ref: true })).toMatchSnapshot()
    expect(t({ titleProp: true, expandProps: true })).toMatchSnapshot()
    expect(t({ ref: true, expandProps: true })).toMatchSnapshot()
  })
})
