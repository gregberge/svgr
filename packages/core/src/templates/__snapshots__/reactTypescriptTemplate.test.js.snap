// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reactTypescriptTemplate should wrap code into a component 1`] = `
"import * as React from 'react'

const Test = () => <Svg />

export default Test"
`;

exports[`reactTypescriptTemplate should wrap code into a component 2`] = `
"import * as React from 'react'

const Test = ({title: string}) => <Svg />

export default Test"
`;

exports[`reactTypescriptTemplate should wrap code into a component 3`] = `
"import * as React from 'react'

const Test = ({svgRef: string}) => <Svg />

export default Test"
`;

exports[`reactTypescriptTemplate should wrap code into a component 4`] = `
"import * as React from 'react'

const Test = (props: any) => <Svg />

export default Test"
`;

exports[`reactTypescriptTemplate should wrap code into a component 5`] = `
"import * as React from 'react'

const Test = ({svgRef: string, title: string}) => <Svg />

export default Test"
`;

exports[`reactTypescriptTemplate should wrap code into a component 6`] = `
"import * as React from 'react'

const Test = ({title: string, ...props}) => <Svg />

export default Test"
`;

exports[`reactTypescriptTemplate should wrap code into a component 7`] = `
"import * as React from 'react'

const Test = ({svgRef: string, ...props}) => <Svg />

export default Test"
`;
